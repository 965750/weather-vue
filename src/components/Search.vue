<template>
    <div class="search my-5 mx-auto">
        <p class="search__label my-0 pl-5">
            {{ $t('Check Your city weather') }}
        </p>
        <form
            id="searchForm"
            @submit.prevent="citySearch"
            class="search__form d-flex"
        >
            <input
                id="searchField"
                v-model="city"
                class="search__field px-5"
                type="text"
                :placeholder="$t('city name')"
            />
            <button
                :disable="!city"
                class="search__submit"
                type="submit"
            >
                {{ $t('search') }}
            </button>
        </form>
        <p
            v-if="getError"
            class="search__error pl-5 caption"
        >
            {{ getError }}
        </p>
    </div>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'Search',
  data() {
    return {
      city: '',
    };
  },
  computed: {
    ...mapGetters(['getError']),
  },
  methods: {
    citySearch() {
      this.$store.dispatch('citySearch', this.city);

      this.city = '';
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../assets/main';

.search {
    width: 300px;

    &__label {
        font-size: 12px;
        color: $aluminium;
    }

    &__form {
        height: 50px;
        overflow: hidden;
    }

    &__field {
        width: 60%;
        border-top-left-radius: 30px;
        border-bottom-left-radius: 30px;
        outline: none;
        background: #FFF;
        line-height: 50px;
        font-size: 14px;
        border: 1px solid $aluminium;

        &::placeholder {
            color: $aluminium;
        }

        &:focus {
            border: 1px solid $deepCerulean;
        }
    }

    &__error {
        color: $persimmon;
    }

    &__submit {
        border-left: none;
        border-top-right-radius: 30px;
        border-bottom-right-radius: 30px;
        width: 40%;
        cursor: pointer;
        background: $deepCerulean;
        outline: none;
        color: #FFF;
        text-transform: uppercase;
        font-size: 20px;

        &:hover {
            background: lighten($deepCerulean, 5%);
        }
    }
}
</style>
