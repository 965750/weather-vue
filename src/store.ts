import Vue from 'vue';
import Vuex from 'vuex';
import axios from 'axios';
import config from '@/config';

import mutations from './mutations';
import getters from './getters';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    lang: 'pl',
    card: null,
    error: null,
  },
  actions: {
    citySearch({ commit }, city: string) {
      axios
        .get(`https://api.openweathermap.org/data/2.5/forecast?q=${city}&cnt=${config.cnt}&APPID=${config.apiKey}`)
        .then((res) => {
          commit('setCard', res.data);
        })
        .catch((err) => {
          commit('setCard__Error', err);
        });
    },
  },
  mutations,
  getters,
});
