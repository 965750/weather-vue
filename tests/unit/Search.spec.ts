import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import VueI18n from 'vue-i18n';
import Vuex from 'vuex';
import Search from '@/components/Search.vue';
import filters from '@/myFilters';
import store from '@/store';
import messages from '@/locale/languages';

const localVue = createLocalVue();
localVue.use(VueI18n);

const initialStore = {
  getters: {
    getError: () => 'We could not find this city name',
  },
  actions: {
    citySearch: jest.fn(),
  },
};

const i18n = new VueI18n({
  locale: store.getters.getLanguage,
  messages,
});

filters.forEach((filter) => {
  localVue.filter(filter.name, filter.handler);
});

describe('Search.vue', () => {
  const createWrapper = (mockStore = initialStore) => shallowMount(Search, {
    store: new Vuex.Store(mockStore),
    i18n,
    localVue,
  });

  it('should match initial snapshot', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match snapshot without error notification', () => {
    const wrapper = createWrapper({
      ...initialStore,
      getters: {
        getError: () => null,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should match make dispatch on submit click', () => {
    const wrapper = createWrapper();

    const input = wrapper.find('#searchField');
    const form = wrapper.find('#searchForm');

    input.element['_value'] = 'karpacz';
    form.trigger('submit');

    expect(initialStore.actions.citySearch).toHaveBeenCalled();
  });
});
