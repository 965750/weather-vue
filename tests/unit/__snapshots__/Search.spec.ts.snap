// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue should match initial snapshot 1`] = `
<div
  class="search my-5 mx-auto"
>
  <p
    class="search__label my-0 pl-5"
  >
    
        Sprawdź pogodę w swoim mieście
    
  </p>
   
  <form
    class="search__form d-flex"
    id="searchForm"
  >
    <input
      class="search__field px-5"
      id="searchField"
      placeholder="nazwa miasta"
      type="text"
    />
     
    <button
      class="search__submit"
      disable="true"
      type="submit"
    >
      
            szukaj
        
    </button>
  </form>
   
  <p
    class="search__error pl-5 caption"
  >
    
        We could not find this city name
    
  </p>
</div>
`;

exports[`Search.vue should match snapshot without error notification 1`] = `
<div
  class="search my-5 mx-auto"
>
  <p
    class="search__label my-0 pl-5"
  >
    
        Sprawdź pogodę w swoim mieście
    
  </p>
   
  <form
    class="search__form d-flex"
    id="searchForm"
  >
    <input
      class="search__field px-5"
      id="searchField"
      placeholder="nazwa miasta"
      type="text"
    />
     
    <button
      class="search__submit"
      disable="true"
      type="submit"
    >
      
            szukaj
        
    </button>
  </form>
   
  <!---->
</div>
`;
